core/src/main/kotlin/entity/Asset.kt

```diff
@@ -0,0 +1,90 @@
+package dev.kord.core.entity
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.KordObject
+import dev.kord.rest.Image
+import dev.kord.rest.route.CdnUrl
+import dev.kord.rest.route.DiscordCdn
+
+public sealed class Asset(
+    public val format: Image.Format,
+    public val cdnUrl: CdnUrl,
+    override val kord: Kord
+) : KordObject {
+    public val url: String
+        get() = cdnUrl.toUrl {
+            this.format = this@Asset.format
+        }
+
+    public val animated: Boolean get() = format is Image.Format.GIF
+
+    public suspend fun getImage(): Image = Image.fromUrl(kord.resources.httpClient, cdnUrl.toUrl())
+
+    public suspend fun getImage(size: Image.Size): Image =
+        Image.fromUrl(kord.resources.httpClient, cdnUrl.toUrl {
+            this.size = size
+        })
+
+    public suspend fun getImage(format: Image.Format): Image =
+        Image.fromUrl(kord.resources.httpClient, cdnUrl.toUrl {
+            this.format = format
+        })
+
+    public suspend fun getImage(format: Image.Format, size: Image.Size): Image =
+        Image.fromUrl(kord.resources.httpClient, cdnUrl.toUrl {
+            this.format = format
+            this.size = size
+        })
+
+    public class Emoji(animated: Boolean, emojiId: Snowflake, kord: Kord) : Asset(if (animated) Image.Format.GIF else Image.Format.WEBP, DiscordCdn.emoji(emojiId), kord)
+
+    public class DefaultUserAvatar(discriminator: Int, kord: Kord) :
+        Asset(Image.Format.PNG, DiscordCdn.defaultAvatar(discriminator), kord)
+
+    public class UserAvatar(userId: Snowflake, avatarHash: String, kord: Kord) :
+        Asset(if (avatarHash.startsWith("a_")) Image.Format.GIF else Image.Format.WEBP, DiscordCdn.userAvatar(userId, avatarHash), kord)
```

Comment:
i think the `hash.startsWith("a_")` logic should be extracted into a function, it's duplicated a lot